src/java/org/apache/cassandra/schema/TableMetadata.java

```diff
@@ -558,7 +558,8 @@ boolean changeAffectsPreparedStatements(TableMetadata updated)
             || !regularAndStaticColumns.equals(updated.regularAndStaticColumns)
             || !indexes.equals(updated.indexes)
             || params.defaultTimeToLive != updated.params.defaultTimeToLive
-            || params.gcGraceSeconds != updated.params.gcGraceSeconds;
+            || params.gcGraceSeconds != updated.params.gcGraceSeconds
+            || flags.contains(Flag.DENSE) && !updated.flags.contains(Flag.DENSE);
```

Comment:
You could use `!Flag.isCQLTable(flags)` instead:  https://github.com/apache/cassandra/blob/61046b5c6836d86f38d238e3629074099a6a6545/src/java/org/apache/cassandra/schema/TableMetadata.java#L99

Reply:
can we also put it in parenthesis for readability…

```suggestion
            || ( !isCQLTable(flags) && isCQLTable(updated.flags) );
```